src/atoms/Heading/Heading.web.js

```diff
@@ -1,5 +1,108 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { getColorKeys, getColor } from '../../_helpers/theme';
+import baseTheme from '../../tokens/theme';
+import automation from '../../_helpers/automation-attributes';
 
-const Heading = () => <div>WIP</div>;
+import Text from '../Text';
+
+const styles = {
+  fontSize({ theme, size }) {
+    switch (size) {
+      case 'medium':
+        return theme.fonts.size.medium;
+      case 'large':
+        return theme.fonts.size.large;
+      case 'xlarge':
+        return theme.fonts.size.xxlarge;
+      case 'xxlarge':
+        return theme.fonts.size.xxxlarge;
+      case 'xxxlarge':
+        return theme.fonts.size.xxxxlarge;
+      default:
+        return theme.fonts.size.xxxxlarge;
+    }
+  },
+  lineHeight({ theme, size }) {
+    switch (size) {
+      case 'medium':
+        return theme.fonts.lineHeight.medium;
+      case 'large':
+        return theme.fonts.lineHeight.medium;
+      case 'xlarge':
+        return theme.fonts.lineHeight.xlarge;
+      case 'xxlarge':
+        return theme.fonts.lineHeight.xxlarge;
+      case 'xxxlarge':
+        return theme.fonts.lineHeight.xxlarge;
+      default:
+        return theme.fonts.lineHeight.xxlarge;
+    }
+  },
+};
+
+const StyledHeading = styled(Text)`
+  &&& {
+    font-family: ${({ theme, _weight }) => theme.fonts.family.lato[_weight]};
+    font-size: ${styles.fontSize};
+    font-weight: ${({ theme, _weight }) => theme.fonts.weight[_weight]};
+    color: ${({ theme, color }) => getColor(theme, color)};
+    letter-spacing: 0;
+    line-height: ${styles.lineHeight};
+  }
+`;
+
+const Heading = ({ size, testID, color, children, maxLines, weight }) => {
+  let headingLevel;
+  switch (size) {
```

Comment:
just create an object outside this component with a map of size to level
```js
const headingLevel = {
'xxxlarge': 'h1',
'xxlarge': 'h2',
}
```
and so on. then you just need to refer it inside the component as 
`as={headingLevel[size]}`

Reply:
But then I'll need to change the size to support the `l_regular`,m_bold etc... but then native has only  support for basic five `sizes`.

Reply:
Aree nai nai my bad.
```
const headingLevel = {
  xxxl: H1,
  xxl: H2,
  xl: H3,
  L: H4,
  M: H5,
};
```